File: hex2dec_v5.1.py
# hexadecimális szám átalakítása decimális számmá (ver 5.1)

def hex2dec(h: str) -> int:
    if h == "0": return 0
    elif h == "1": return 1
    elif h == "2": return 2
    elif h == "3": return 3
    elif h == "4": return 4
    elif h == "5": return 5
    elif h == "6": return 6
    elif h == "7": return 7
    elif h == "8": return 8
    elif h == "9": return 9
    elif h in ("a", "A"): return 10
    elif h in ("b", "B"): return 11
    elif h in ("c", "C"): return 12
    elif h in ("d", "D"): return 13
    elif h in ("e", "E"): return 14
    elif h in ("f", "F"): return 15
    else:
        raise ValueError("Hibás számjegy: " + h)

x = "24C"

print("Hexadecimal number:", x)
print()

d = 0
h = 1

# helyiérték meghatározása (16^(hossz-1))
i = 1
while i < len(x):
    h *= 16
    i += 1

# hex szám feldolgozása
i = 0
while i < len(x):
    sz = hex2dec(x[i])
    r = sz * h
    d += r
    h //= 16
    i += 1

print("Decimal number:", d)
print("__________")

File: hex2dec_v5.1.js
// hexadecimális szám átalakítása decimális számmá (ver 5.1)

function hex2dec(h) {
  let sz;
  switch (h) {
    case "0": sz = 0; break;
    case "1": sz = 1; break;
    case "2": sz = 2; break;
    case "3": sz = 3; break;
    case "4": sz = 4; break;
    case "5": sz = 5; break;
    case "6": sz = 6; break;
    case "7": sz = 7; break;
    case "8": sz = 8; break;
    case "9": sz = 9; break;
    case "a":
    case "A": sz = 10; break;
    case "b":
    case "B": sz = 11; break;
    case "c":
    case "C": sz = 12; break;
    case "d":
    case "D": sz = 13; break;
    case "e":
    case "E": sz = 14; break;
    case "f":
    case "F": sz = 15; break;
    default:
      throw new Error("Hibás számjegy: " + h);
  }
  return sz;
}

let x = "24C";

console.log("Hexadecimal number: " + x);
console.log();

let d = 0;
let h = 1;

// helyiérték meghatározása (16^(hossz-1))
let i = 1;
while (i < x.length) {
  h *= 16;
  i++;
}

// hex szám feldolgozása
i = 0;
while (i < x.length) {
  let sz = hex2dec(x[i]);
  let r = sz * h;
  d += r;
  h /= 16;
  i++;
}

console.log("Decimal number: " + d);
console.log("__________");
